Chat_App/Core/Custom UI/TextView/TextView.swift

```diff
@@ -0,0 +1,160 @@
+        containerView.layer.borderWidth = Constants.ContainerView.borderWidth
+        containerView.layer.borderColor = Constants.Colors.borderColor
+        containerView.layer.cornerRadius = Constants.ContainerView.cornerRadius
+        containerView.translatesAutoresizingMaskIntoConstraints = false
+        
+    }
+    
+    private func setUpButton() {
+        sendButton.setImage(UIImage(named: Constants.AssetIdentifier.sendButton.rawValue), for: .normal)
+        sendButton.addTarget(self, action: #selector(sendMessage), for: .touchUpInside)
+        sendButton.translatesAutoresizingMaskIntoConstraints = false
+        
+    }
+    
+    @objc func sendMessage() {
+        delegate?.buttonTapped(sender: sendButton)
+    }
+    //MARK: - Constraints
+    func setUpConstraints() {
+        
+        NSLayoutConstraint.activate([
+            containerView.topAnchor.constraint(equalTo: topAnchor),
+            containerView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            containerView.bottomAnchor.constraint(equalTo: bottomAnchor),
+            containerView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            containerView.heightAnchor.constraint(equalToConstant: Constants.TextView.height)
+        ])
+        
+        NSLayoutConstraint.activate([
+            textView.topAnchor.constraint(equalTo: containerView.topAnchor, constant: Constants.TextView.top),
+            textView.bottomAnchor.constraint(equalTo: containerView.bottomAnchor, constant: Constants.TextView.bottom),
+            textView.leadingAnchor.constraint(equalTo: containerView.leadingAnchor),
+            textView.trailingAnchor.constraint(equalTo: sendButton.leadingAnchor)
+            
+        ])
+        NSLayoutConstraint.activate([
+            placeholder.centerYAnchor.constraint(equalTo: containerView.centerYAnchor),
+            placeholder.leadingAnchor.constraint(equalTo: containerView.leadingAnchor, constant: Constants.TextView.leftEdge)
+        ])
+        NSLayoutConstraint.activate([
+            sendButton.centerYAnchor.constraint(equalTo: containerView.centerYAnchor),
+            sendButton.trailingAnchor.constraint(equalTo: containerView.trailingAnchor, constant: -Constants.SendButton.trailing)
+        ])
+        
+    }
+}
+//MARK: - UITextViewDelegate
+extension TextView: UITextViewDelegate {
+    func textViewDidBeginEditing(_ textView: UITextView) {
+        placeholder.isHidden = true
+    }
+    
+    func textViewDidEndEditing(_ textView: UITextView) {
+        if textView.text.isEmpty {
+            placeholder.isHidden = false
+        }
+    }
+    func textViewDidChange(_ textView: UITextView) {
+        
+        let fixedWidth = textView.frame.size.width
+        let newSize = textView.sizeThatFits(CGSize(width: fixedWidth, height: .infinity))
+        let newHeight = min(newSize.height, Constants.TextView.maxNumberOfLines * lineHeight)
+        textView.constraints.forEach { constraint in
```

Comment:
this is pretty invasive method of changing constraint's properties. Better way would be to store the height constraint like:
private lazy var textViewHeightConstraint = textView.heightAnchor.constraint(....)

then activate it somewhere in your setup methods. Then change it's constant right here and `UIView.animate` with `layoutIfNeeded()` call inside if you need to animate these height changes